gulpfile.js

```diff
@@ -437,7 +394,55 @@ gulp.task('generate-icon-names', () => {
       }),
     )
     .pipe(gulp.dest(path.join(PATHS.output, 'Icon', 'Classes', 'Generated')));
+  done();
+});
+
+gulp.task(
+  'template',
+  gulp.series('generate-icon-names', () => {
+    const streams = [];
+    const templateData = parseTokens(tokens);
+
+    // eslint-disable-next-line no-restricted-syntax
+    for (const type of TYPES) {
+      const processedType = format(type);
+
+      streams.push(
+        gulp
+          .src(path.join(PATHS.templates, `BPK${processedType}.h.njk`))
+          .pipe(data(() => templateData))
+          .pipe(nunjucks.compile())
+          .pipe(
+            rename(`${processedType}/Classes/Generated/BPK${processedType}.h`),
+          ),
+      );
+
+      streams.push(
+        gulp
+          .src(path.join(PATHS.templates, `BPK${processedType}.m.njk`))
+          .pipe(data(() => templateData))
+          .pipe(nunjucks.compile())
+          .pipe(
+            rename(`${processedType}/Classes/Generated/BPK${processedType}.m`),
+          ),
+      );
+    }
+
+    return merge2(streams).pipe(gulp.dest(PATHS.output));
+  }),
+);
+
+gulp.task('copy-icon-font', done => {
```

Comment:
Gulp 4 now requires that we signal async completion

Reply:
this is done by invoking `done` (where `done` is the first argument of the callback)

Reply:
So merge2 is blocking here?